src/middlewared/middlewared/plugins/interface/netif_linux/vlan.py

```diff
@@ -1,40 +1,32 @@
-# -*- coding=utf-8 -*-
-import glob
-import logging
-import os
-import re
-import subprocess
-
-import middlewared.plugins.interface.netif_linux.interface as interface
+from glob import glob
+from os.path import basename
+from os import readlink
+from logging import getLogger
+from re import search
+from pyroute2 import NDB
 
 from .utils import run
 
-logger = logging.getLogger(__name__)
+logger = getLogger(__name__)
 
 __all__ = ["create_vlan", "VlanMixin"]
 
 
 def create_vlan(name, parent, tag):
-    try:
-        run(["ip", "link", "add", "link", parent, "name", name, "type", "vlan", "id", str(tag)])
-    except subprocess.CalledProcessError as e:
-        if e.stderr.startswith("Cannot find device "):
-            raise FileNotFoundError(e.stderr)
-
-        raise
-
-    interface.Interface(name).up()
+    with NDB(log="off") as ndb:
+        ndb.interfaces[parent].set("state", "up").commit()  # make sure parent is up
+        ndb.interfaces.create(ifname=name, link=parent, vlan_id=tag, kind="vlan").set("state", "up").commit()
 
 
 class VlanMixin:
     @property
     def parent(self):
-        return os.path.basename(os.readlink(glob.glob(f"/sys/devices/virtual/net/{self.name}/lower_*")[0]))
+        return basename(readlink(glob(f"/sys/devices/virtual/net/{self.name}/lower_*")[0]))
 
     @property
     def tag(self):
         with open(f"/proc/net/vlan/{self.name}") as f:
-            return int(re.search(r"VID: ([0-9]+)", f.read()).group(1))
+            return int(search(r"VID: ([0-9]+)", f.read()).group(1))
```

Comment:
What's the point of all these changes?

Reply:
no point in importing the entire module when only using singular functions.

Reply:
@yocalebo also no point in introducing unnecessary diffs, no point in confusing developers (what is `search`?), no point in making us change `import` line every time we want to use a new function from common library

"import os" - 24,732,052 code results - https://github.com/search?q=%22import+os%22&type=code
"from os import" - 218,819 code results - https://github.com/search?q=%22from+os+import%22&type=code

Reply:
If a developer gets confused because they cant see that the `search` wasn't imported from `re` then we have bigger problems. Furthermore, linking a github search is relevant in what way exactly? Why does it matter what other people do? Our entire middlewared code base does things that aren't "standard" so those queries are superfluous.

Reply:
But alas, I'm not really interested in arguing with you on what perceive to be the "right" way of doing things so I've removed the "unnecessary diffs".